daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/UpgradesSpec.scala

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.validation
+
+import com.daml.integrationtest.CantonFixture
+import org.scalatest.Inside
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpec
+
+import scala.concurrent.{Future}
+import com.google.protobuf.ByteString
+import com.daml.bazeltools.BazelRunfiles
+import java.io.File
+import java.io.FileInputStream
+import org.scalatest.Suite
+import org.scalatest.compatible.Assertion
+
+import scala.io.Source
+import com.daml.lf.data.Ref.PackageId
+
+import com.daml.lf.archive.{DarReader}
+import scala.util.{Success, Failure}
+
+final class UpgradesIT extends AsyncWordSpec with Matchers with Inside with CantonFixture {
+  self: Suite =>
+  override lazy val devMode = true;
+  override val cantonFixtureDebugMode = true;
+  override val cantonFixtureDebugModeRemoveTmpFilesRegardless = true;
```

Comment:
Since `cantonFixtureDebugModeRemoveTmpFilesRegardless` being true is meaningless without `cantonFixtureDebugMode`, I wonder if this should be an enum of some kind

Reply:
Afaict, only two things actually use `cantonFixtureDebugMode`, so I can do this without making this PR too much bigger. On it

Reply:
fixed